src/hooks/useRemoteImage.test.ts

```diff
@@ -0,0 +1,41 @@
+import * as RD from '@devexperts/remote-data-ts'
+
+import { useRemoteImage } from './useRemoteImage'
+
+const mockSetValue = jest.fn()
+
+jest.mock('react', () => ({
+  useState: (value: unknown) => [value, mockSetValue]
+}))
+
+describe('hooks/useRemoteImage', () => {
+  const LOAD_FAILURE_SRC = 'fail'
+  const LOAD_SUCCESS_SRC = 'success'
+
+  beforeEach(() => {
+    // Mocking Image.prototype.src to call the onload or onerror
+    Object.defineProperty(global.Image.prototype, 'src', {
+      set(src: string) {
+        if (src === LOAD_FAILURE_SRC) {
+          this.onerror()
+        } else if (src === LOAD_SUCCESS_SRC) {
+          this.onload()
+        }
+      }
+    })
+  })
```

Comment:
As with other mocks or stubs, I would recommend to bring `Image` back to it's original "state" in `afterAll` or so.

Reply:
@veado nice catch